library/src/blas3/Tensile/gemm.cpp

```diff
@@ -28,20 +23,20 @@ namespace
     * GEMM implementation
     ******************************************************************************/
     template <typename T>
-    rocblas_status rocblas_gemm_impl(rocblas_handle    handle,
-                                     rocblas_operation trans_a,
-                                     rocblas_operation trans_b,
-                                     rocblas_int       m,
-                                     rocblas_int       n,
-                                     rocblas_int       k,
-                                     const T*          alpha,
-                                     const T*          A,
-                                     rocblas_int       ld_a,
-                                     const T*          B,
-                                     rocblas_int       ld_b,
-                                     const T*          beta,
-                                     T*                C,
-                                     rocblas_int       ld_c)
+    auto rocblas_gemm_impl(rocblas_handle    handle,
```

Comment:
Overuse of auto may reduce readability.

Reply:
> Overuse of auto may reduce readability.

I don't know how that got in there -- when the return type of a function is a simple name like `rocblas_status`, it should be used. It's only when the type is very long or would require `decltype()` that `auto` may simplify things.